File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        
        if(s.length() == 1)
            return 1;
        
        Set<Character> set= new HashSet<>();
        
        for(char c : s.toCharArray()){
            if(!set.add(c)){
                set.remove(c);
            } else {
                set.add(c);
            }
        }
        
        int count = s.length() - set.size();
        
        if(set.size() > 0){
            count += 1;
        }
        
        return count;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        
        letters = set()
        count = 0

        for c in s:
            if c in letters:
                letters.remove(c)
                count += 2
            else:
                letters.add(c)

        if len(letters) > 0:
            count += 1  

        return count